client-core/src/synchronizer/auto_sync.rs

```diff
@@ -71,6 +73,21 @@ impl AutoSync {
         self.send_json(data)
     }
 
+    /// Get sync information
+    pub fn sync_info(&self) -> Result<AutoSyncInfo> {
+        let mut ret = AutoSyncInfo::default();
+
+        {
+            let data = self.data.lock().expect("get progress autosync lock");
```

Comment:
instead of copying everything here, what if there's `info: AutoSyncInfo` field in `self.data`, so one just do something like:
```
let data = self.data.lock().expect("get progress autosync lock");
data.info.clone()
```
?

Reply:
ok